components/TARGET_PSA/services/psa_prot_internal_storage/COMPONENT_PSA_SRV_IMPL/pits_impl.cpp

```diff
@@ -48,52 +48,88 @@ const uint8_t base64_coding_table[] = {
     '4', '5', '6', '7', '8', '9', '+', '-'
 };
 
-
+/*
+ * \brief Get default KVStore instance for internal flesh storage
+ *
+ * \return valid pointer to KVStore
+ */
 static KVStore *get_kvstore_instance(void)
 {
     KVMap &kv_map = KVMap::get_instance();
 
-    return kv_map.get_main_kv_instance(STR_EXPAND(MBED_CONF_STORAGE_DEFAULT_KV));
+    KVStore *kvstore = kv_map.get_internal_kv_instance(STR_EXPAND(MBED_CONF_STORAGE_DEFAULT_KV));
+    if (!kvstore) {
+        // Can only happen due to system missconfiguration.
+        // Thus considered as unrecoverable error for runtime.
+        error("Failed getting kvstore instance\n");
+    }
+    return kvstore;
+}
+
+/*
+ * \brief Convert KVStore stauts codes to PSA internal storage status codes
+ *
+ * \param[in] status - KVStore status code
+ * \return PSA internal storage status code
+ */
+static psa_its_status_t convert_status(int status)
+{
+    switch (status) {
+        case MBED_SUCCESS:
+            return PSA_ITS_SUCCESS;
+        case MBED_ERROR_WRITE_PROTECTED:
+            return PSA_ITS_ERROR_WRITE_ONCE;
+        case MBED_ERROR_MEDIA_FULL:
+            return PSA_ITS_ERROR_INSUFFICIENT_SPACE;
+        case MBED_ERROR_ITEM_NOT_FOUND:
+            return PSA_ITS_ERROR_KEY_NOT_FOUND;
+        default:
+            return PSA_ITS_ERROR_STORAGE_FAILURE;
+    }
 }
 
-static void generate_fn(char *tdb_filename, uint32_t tdb_file_len, uint32_t uid, uint32_t pid)
+/*
+ * \breif Generate KVStore file name
+ *
+ * Generate KVStore by Base64 encoding PID and UID with a delimiter
+ * Delimiter is required for determining between PID and UID
+ *
+ * \param[out] tdb_filename - pointer to a buffer for the file name
+ * \param[in]  tdb_filename_size - output buffer size
+ * \param[in]  uid - PSA internal storage unique ID
+ * \param[in]  pid - owner PSA partition ID
+ */
+static void generate_fn(char *tdb_filename, uint32_t tdb_filename_size, uint32_t uid, uint32_t pid)
 {
     MBED_ASSERT(tdb_filename != NULL);
-    MBED_ASSERT(tdb_file_len >= PSA_ITS_FILENAME_MAX_LEN);
+    MBED_ASSERT(tdb_filename_size == PSA_ITS_FILENAME_MAX_LEN);
 
     uint8_t filename_idx = 0;
-    uint32_t tmp_uid = uid;
-    uint32_t tmp_pid = pid;
 
-    // Iterate on UID; each time convert 6 bits of UID into a character; first iteration must be done
+    // Iterate on PID; each time convert 6 bits of PID into a character; first iteration must be done
```

Comment:
to prevent possibility of timing attacks.
offending partition(NSPE) may try to read UIDs belonging to other partitions, our storage back-end does not implement constant time comparisons, thus this for determining the file is missing and the time determining that file is accessible may vary.
 in case PID comes first - the lookup will be started by a PID and offending partitions could not leverage timing attacks.